api/lib/domain/project/createSource.ts

```diff
@@ -12,6 +12,10 @@ interface UploadSourceCommand {
 }
 
 export const createSource = factory<Project, UploadSourceCommand, Source>(function (project, command, emitter) {
+  if (command.columns.findIndex(column => !column || column.trim() === '') > 0) {
```

Comment:
This is a nit-pick, but I would probably use `.some(...)` instead of `.findIndex(...) > 0`.

Reply:
Good point, why not.